packages/frontend/src/components/NoteComposer.tsx

```
@@ -85,118 +95,98 @@ export default function NoteComposer({
+            </TabsContent>
+
+            <TabsContent value="preview" className="mt-3">
+              <div className="min-h-[76px] px-3 py-2 bg-muted border rounded-md">
+                {isLoadingPreview ? (
+                  <p className="text-muted-foreground italic">
+                    Loading preview...
+                  </p>
+                ) : previewHtml ? (
+                  <div
+                    className="prose prose-sm dark:prose-invert max-w-none"
+                    dangerouslySetInnerHTML={{ __html: previewHtml }}
```

Biome (2.3.13)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)